WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -86,10 +89,15 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 
 				foreach (var finishedCoinJoin in finishedCoinJoins)
 				{
-					var removed = trackedWallets.Remove(finishedCoinJoin.Wallet.WalletName);
+					var walletToRemove = finishedCoinJoin.Wallet;
+					var removed = trackedWallets.Remove(walletToRemove.WalletName);
 					if (!removed)
 					{
-						Logger.LogWarning($"Wallet: `{finishedCoinJoin.Wallet.WalletName}` was not removed from tracked wallet list. Will retry in a few seconds.");
+						Logger.LogWarning($"Wallet: `{walletToRemove.WalletName}` was not removed from tracked wallet list. Will retry in a few seconds.");
+					}
+					else
+					{
+						WalletStatusChanged?.Invoke(this, new WalletStatusChangedEventArgs(walletToRemove, false));
```

Comment:
To make it more clear:
```suggestion
						WalletStatusChanged?.Invoke(this, new WalletStatusChangedEventArgs(walletToRemove, IsCoinJoining: false));
```